File: zhumaysymba/front/pages/api/models/employee.ts
import {Badge} from "./badge";
import {Task} from "./task";
import {Gift} from "./gift";

export interface Employee {
    ID: number;
    Name: string;
    Balance: string;
    About: string;
    RocketChatID: string;
    ImageUrl: string;

    Badges: Badge[];
    Tasks:  Task[];
    Gifts:  Gift[];
}

File: zhumaysymba/back/internal/entity/employee.go
package entity

import (
	"github.com/somnoynadno/zhumaysynba/internal/base"
)

type Employee struct {
	base.Entity

	Name     string
	Balance  uint
	About    string
	ImageUrl string

	RocketChatID string

	EmployeeUnits []EmployeeUnits
	Tasks         []Task  `gorm:"many2many:employee_tasks"`
	BoughtGifts   []Gift  `gorm:"many2many:bought_gifts"`
	Badges        []Badge `gorm:"many2many:employee_badges"`
}
